### PR TITLE
Make docs left navigation scannable

### PR DESCRIPTION
**Pull request by eriwen**
This changes the hierarchy of docs pages such that it becomes more obvious where to find desired pages without search. Do this by breaking up long lists of links and improving categorization by use-case.

* Add Tutorials and Integrating Gradle subsections
* Merge Continuous Build into CLI docs
* Segment authoring builds sections more by project language

### Context
I want to get us closer to the proposed longer-term structure before we make guides have the same navigation.

### Before
screenshot 2018-02-14 16 17 57

### After
screenshot 2018-02-14 16 18 10

### CODE REVIEW

File: subprojects/docs/src/docs/stylesheets/userGuideHtml.xsl [hunk 1]
```diff
@@ -52,171 +52,207 @@
...
-                            <li><a href="/userguide/build_cache.html">Build Cache</a></li>
-                            <li><a href="/userguide/composite_builds.html">Composite Builds</a></li>
-                            <li><a href="/userguide/continuous_build.html">Continuous Build</a></li>
-                            <li><a href="/userguide/gradle_daemon.html">Daemon</a></li>
                             <li><a href="/userguide/troubleshooting.html">Troubleshooting</a></li>
-                            <li><a href="/userguide/embedding.html">Embedding Gradle</a></li>
                         </ul>
 
-                        <h3 id="writing-gradle-builds">Authoring Gradle Builds</h3>
+                        <h3 id="tutorials">Project Tutorials</h3>
```
Comment (bmuschko) [inline subprojects/docs/src/docs/stylesheets/userGuideHtml.xsl]: Should we rather say "Language Tutorials". I am not quite sure what "Project" is supposed to mean.

**Comment by oehme** [comment]
I like it! I think the CLI should be before the build cache though. I know we're proud of it, but it's probably not the first thing you'll need to understand.

File: subprojects/docs/src/docs/stylesheets/userGuideHtml.xsl (hunk 1 above)
Comment (eriwen) [reply subprojects/docs/src/docs/stylesheets/userGuideHtml.xsl]: It's supposed to mean that you build a project. Not sure how to tersely make that clear.

Comment (eriwen) [reply subprojects/docs/src/docs/stylesheets/userGuideHtml.xsl]: Alright, I wrote `Project-Based Tutorials` here

**Comment by oehme** [comment]
Why remove continuous build? It's super useful and already less known than it should be.

**Comment by eriwen** [comment]
Thanks fellas. This is much better. @oehme To me the subsections of the nav are sorta ordered in what to start with before moving on. Each subsection internal is alphabetically sorted so that it's easier to find something you're looking for.

**Comment by eriwen** [comment]
@oehme Continuous build is now contained within CLI chapter

**Comment by eriwen** [comment]
BTW composite builds moved under "Authoring" because it really was about configuring it, not using a composite build.

**Pull request merged**